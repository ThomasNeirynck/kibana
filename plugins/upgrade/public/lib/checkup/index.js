import {
  getFromApi,
} from '../request';


export async function getDeprecations() {
  return await getFromApi(`/api/migration/deprecations`);
}
