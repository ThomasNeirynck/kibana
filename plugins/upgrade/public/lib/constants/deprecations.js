export const DEPRECATION_ISSUE_LEVELS = [
  'critical',
  'warning',
  'info',
  'none',
];
