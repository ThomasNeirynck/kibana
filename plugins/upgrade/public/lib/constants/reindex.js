export const REINDEX_STEP_SEQUENCE = [
  'CREATE_INDEX',
  'SET_READONLY',
  'REINDEX',
  'REFRESH_INDEX',
  'VERIFY_DOCS',
  'REPLACE_INDEX',
];

export const REINDEX_STEPS = REINDEX_STEP_SEQUENCE
  .reduce((steps, step) => ({
    ...steps,
    [step]: step,
  }), {});

export const UPGRADE_STEP_SEQUENCE = [
  'UPGRADE',
];

export const UPGRADE_STEPS = UPGRADE_STEP_SEQUENCE
  .reduce((steps, step) => ({
    ...steps,
    [step]: step,
  }), {});

export const NUMBER_OF_STEPS = {
  REINDEX: REINDEX_STEP_SEQUENCE.length,
  UPGRADE: UPGRADE_STEP_SEQUENCE.length,
};

export const STEP_RESULTS = {
  COMPLETED:   'COMPLETED',
  FAILED:      'FAILED',
  RUNNING:     'RUNNING',
  NOT_STARTED: 'NOT_STARTED',
  CANCELED:    'CANCELED',
};

export const INDEX_ACTION = {
  TYPE: {
    REINDEX: 'REINDEX',
    UPGRADE: 'UPGRADE',
  },
  LABEL: {
    REINDEX: 'Reindex',
    UPGRADE: 'Upgrade',
  },
};
