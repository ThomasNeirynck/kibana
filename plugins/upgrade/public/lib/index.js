export { timeout } from './util';
export {
  getAssistance,
  getMappingsAndSettings,
  createIndex,
  setReadOnly,
  runReindex,
  getTaskDetails,
  refreshIndex,
  verifyDocs,
  getSettingsAndAliases,
  updateRefreshInterval,
  replaceIndex,
  deleteTask,
  resetIndex,
  cancelTask,
  runUpgrade,
  isCompleted,
  isFailed,
  isNotStarted,
  isRunning,
  isStepCompleted,
  isStepFailed,
  isStepNotStarted,
  isStepRunning,
  isResettable,
  getStepMessage,
  wrapErrorMessage,
} from './reindex';
