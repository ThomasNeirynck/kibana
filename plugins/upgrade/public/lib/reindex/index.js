export {
  isCompleted,
  isFailed,
  isNotStarted,
  isRunning,
  isCanceled,
  isResettable,
  isCancelable,
  isStepCompleted,
  isStepFailed,
  isStepNotStarted,
  isStepRunning,
  isStepCanceled,
} from './state';

export {
  getStepMessage,
  wrapErrorMessage,
} from './messages';

export {
  getAssistance,
  getMappingsAndSettings,
  getSettingsAndAliases,
  updateRefreshInterval,
  createIndex,
  setReadOnly,
  runReindex,
  getTaskDetails,
  cancelTask,
  refreshIndex,
  verifyDocs,
  replaceIndex,
  deleteTask,
  resetIndex,
  runUpgrade,
} from './step_methods';

