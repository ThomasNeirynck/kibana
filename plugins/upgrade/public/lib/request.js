import chrome from 'ui/chrome';

function get(url) {
  return new Promise((resolve, reject) => {
    const req = new window.XMLHttpRequest();

    req.open('GET', url);

    req.onload = () => {
      if (req.status === 200) {
        resolve(JSON.parse(req.response));
      } else {
        reject(JSON.parse(req.response));
      }
    };

    req.onerror = () => {
      reject(new Error('Network error'));
    };

    req.send();
  });
}

export function getFromApi(url) {
  return get(chrome.addBasePath(url));
}

function post(url, data) {
  return new Promise((resolve, reject) => {
    const xhr = new window.XMLHttpRequest();
    xhr.open('POST', url);
    xhr.setRequestHeader('Content-type', 'application/json');
    xhr.setRequestHeader('kbn-version', chrome.getXsrfToken());
    xhr.onload = () => {
      if (xhr.status === 200) {
        resolve(JSON.parse(xhr.response));
      } else {
        reject(JSON.parse(xhr.response));
      }
    };
    xhr.send(JSON.stringify(data));
  });
}

export function postToApi(url, data) {
  return post(chrome.addBasePath(url), data);
}

function put(url, data) {
  return new Promise((resolve, reject) => {
    const xhr = new window.XMLHttpRequest();
    xhr.open('PUT', url);
    xhr.setRequestHeader('Content-type', 'application/json');
    xhr.setRequestHeader('kbn-version', chrome.getXsrfToken());
    xhr.onload = () => {
      if (xhr.status === 200) {
        resolve(JSON.parse(xhr.response));
      } else {
        reject(JSON.parse(xhr.response));
      }
    };
    xhr.send(JSON.stringify(data));
  });
}

export function putToApi(url, data) {
  return put(chrome.addBasePath(url), data);
}

function del(url) {
  return new Promise((resolve, reject) => {
    const xhr = new window.XMLHttpRequest();
    xhr.open('DELETE', url);
    xhr.setRequestHeader('Content-type', 'application/json');
    xhr.setRequestHeader('kbn-version', chrome.getXsrfToken());
    xhr.onload = () => {
      if (xhr.status === 200) {
        resolve(JSON.parse(xhr.response));
      } else {
        reject(JSON.parse(xhr.response));
      }
    };
    xhr.send();
  });
}

export function deleteFromApi(url) {
  return del(chrome.addBasePath(url));
}
