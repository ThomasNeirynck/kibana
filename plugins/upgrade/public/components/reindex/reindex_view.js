import PropTypes from 'prop-types';
import React from 'react';

import { IndexTable } from './index_table';
import { ReindexInfo } from './info';
import { InfoGroup } from '../info_group';
import { withViewState } from '../../lib/util/view_state';
import { ErrorPanel } from '../error_panel';
import { LOADING_STATUS } from '../../lib/constants';


export const ReindexView = withViewState({
  initialState: {
    isInfoCollapsed: false,
  },
  updaters: {
    toggleInfoCollapsed: (state) => () => ({
      isInfoCollapsed: !state.isInfoCollapsed,
    }),
  },
})(function ReindexView({
  cancelAction,
  indices,
  isInfoCollapsed,
  loadIndices,
  resetAction,
  processIndex,
  toggleInfoCollapsed,
  loadingStatus,
  errorMessage,
}) {
  return (
    <div className="kuiView">
      <div className="kuiViewContent kuiViewContent--constrainedWidth">
        <div className="kuiViewContentItem">
          <InfoGroup
            className="kuiVerticalRhythm"
            isCollapsed={ isInfoCollapsed }
            onChangeCollapsed={ toggleInfoCollapsed }
            title="Reindex Helper"
          >
            <ReindexInfo className="kuiVerticalRhythm" />
          </InfoGroup>

          {
            loadingStatus === LOADING_STATUS.FAILURE
              ? (
                <ErrorPanel className="kuiVerticalRhythm">
                  { errorMessage }
                </ErrorPanel>
              )
              : null
          }

          <IndexTable
            cancelAction={ cancelAction }
            className="kuiVerticalRhythm"
            indices={ indices }
            loadIndices={ loadIndices }
            processIndex={ processIndex }
            resetAction={ resetAction }
          />
        </div>
      </div>
    </div>
  );
});

ReindexView.propTypes = {
  cancelAction: PropTypes.func,
  indices: PropTypes.objectOf(PropTypes.object),
  isInfoCollapsed: PropTypes.bool,
  loadIndices: PropTypes.func,
  resetAction: PropTypes.func,
  processIndex: PropTypes.func,
  toggleInfoCollapsed: PropTypes.func,
  loadingStatus: PropTypes.string,
  errorMessage: PropTypes.node,
};

ReindexView.defaultProps = {
  cancelAction: () => {},
  indices: {},
  loadIndices: () => {},
  resetAction: () => {},
  processIndex: () => {},
  loadingStatus: '',
  errorMessage: '',
};

