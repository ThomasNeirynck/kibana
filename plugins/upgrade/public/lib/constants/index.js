import { HomeView } from '../../components/home';
import { CheckupView } from '../../components/checkup';
import { ReindexView } from '../../components/reindex';
import { LoggingView } from '../../components/logging';


export {
  NUMBER_OF_STEPS,
  REINDEX_STEPS,
  UPGRADE_STEPS,
  STEP_RESULTS,
  INDEX_ACTION,
} from './reindex';

export {
  DEPRECATION_ISSUE_LEVELS,
} from './deprecations';

export const VIEWS = {
  HOME: {
    component: HomeView,
    label: 'Overview',
    location: 'home',
  },
  CHECKUP: {
    component: CheckupView,
    label: 'Cluster Checkup',
    location: 'checkup',
  },
  REINDEX: {
    component: ReindexView,
    label: 'Reindex Helper',
    location: 'reindex',
  },
  LOGGING: {
    component: LoggingView,
    label: 'Toggle Deprecation Logging',
    location: 'logging',
  }
};
export const DEFAULT_VIEW_ID = 'HOME';

export const HUMAN_READABLE_DELAY = 750;

export const LOADING_STATUS = {
  FORBIDDEN: 'FORBIDDEN',
  FAILURE: 'FAILURE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  UNINITIALIZED: 'UNINITIALIZED',
};
