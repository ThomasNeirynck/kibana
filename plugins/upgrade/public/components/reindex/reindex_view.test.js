import _ from 'lodash';
import { shallow } from 'enzyme';
import React from 'react';

import { ReindexView } from './reindex_view';
import { LOADING_STATUS } from '../../lib/constants';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function Notifier() {},
}), { virtual: true });


describe('ReindexView', () => {
  test('renders an expanded InfoGroup by default', () => {
    const component = (
      <ReindexView
        setViewState={_.noop}
        viewState={{}}
      />
    );

    const wrapper = shallow(component).dive();

    expect(wrapper.find('InfoGroup').prop('isCollapsed')).toBe(false);
  });

  test('renders a collapsed InfoGroup when the viewState declares it', () => {
    const component = (
      <ReindexView
        setViewState={_.noop}
        viewState={{
          ReindexView: {
            isInfoCollapsed: true,
          },
        }}
      />
    );

    const wrapper = shallow(component).dive();

    expect(wrapper.find('InfoGroup').prop('isCollapsed')).toBe(true);
  });

  test('calls setViewState when the InfoPanel\'s collapse button is clicked' , () => {
    const setViewState = jest.fn();
    const component = (
      <ReindexView
        setViewState={setViewState}
        viewState={{}}
      />
    );

    const wrapper = shallow(component).dive();
    wrapper.find('InfoGroup').prop('onChangeCollapsed')();

    expect(setViewState).toHaveBeenCalledTimes(1);
  });

  test('renders an IndexTable when loadingStatus indicates success', () => {
    const cancelAction = () => {};
    const indices = {
      index1: {},
    };
    const loadIndices = () => {};
    const processIndex = () => {};
    const resetAction = () => {};
    const component = (
      <ReindexView
        loadingStatus={LOADING_STATUS.SUCCESS}
        cancelAction={cancelAction}
        indices={indices}
        loadIndices={loadIndices}
        processIndex={processIndex}
        resetAction={resetAction}
        setViewState={_.noop}
        viewState={{}}
      />
    );

    const wrapper = shallow(component).dive();
    expect(wrapper.find('IndexTable').props()).toMatchObject({
      cancelAction,
      indices,
      loadIndices,
      processIndex,
      resetAction,
    });
  });

  test('renders an ErrorPanel when the loadingStatus indicates forbidden access', () => {
    const component = (
      <ReindexView
        loadingStatus={LOADING_STATUS.FORBIDDEN}
        setViewState={_.noop}
        viewState={{}}
      />
    );

    const wrapper = shallow(component).dive();
    expect(wrapper.find('ErrorPanel')).toMatchSnapshot();
  });

  test('renders an ErrorPanel with the errorMessage when the loadingStatus indicates a failure', () => {
    const component = (
      <ReindexView
        errorMessage="an error message"
        loadingStatus={LOADING_STATUS.FAILURE}
        setViewState={_.noop}
        viewState={{}}
      />
    );

    const wrapper = shallow(component).dive();
    expect(wrapper.find('ErrorPanel')).toMatchSnapshot();
  });
});
