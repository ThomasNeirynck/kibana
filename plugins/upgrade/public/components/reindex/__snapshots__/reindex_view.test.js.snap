// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReindexView renders an ErrorPanel with the errorMessage when the loadingStatus indicates a failure 1`] = `
<ErrorPanel
  className="kuiVerticalRhythm"
  classNames=""
  title="An error occurred."
>
  an error message
</ErrorPanel>
`;
